x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -43,168 +49,189 @@ function ActionMenuButton({ onClick }: { onClick: () => void }) {
   );
 }
 
+function removeEmptyItems<T extends {}>(
+  items: Array<T | null | undefined>
+): T[] {
+  return items.filter(item => !!item) as T[];
+}
+
 interface Props {
   readonly transaction: Transaction;
 }
 
-interface State {
-  readonly isOpen: boolean;
-}
+export const TransactionActionMenu: FunctionComponent<Props> = (
+  props: Props
+) => {
+  const { transaction } = props;
 
-export class TransactionActionMenu extends React.Component<Props, State> {
-  public state: State = {
-    isOpen: false
-  };
+  const [isOpen, setIsOpen] = useState(false);
 
-  public toggle = () => {
-    this.setState(state => ({ isOpen: !state.isOpen }));
-  };
-
-  public close = () => {
-    this.setState({ isOpen: false });
-  };
+  const { urlParams } = useUrlParams();
 
-  public getInfraActions() {
-    const { transaction } = this.props;
-    const hostName = idx(transaction, _ => _.host.hostname);
-    const podId = idx(transaction, _ => _.kubernetes.pod.uid);
-    const containerId = idx(transaction, _ => _.container.id);
-    const traceId = idx(transaction, _ => _.trace.id);
-    const time = new Date(transaction['@timestamp']).getTime();
-    const infraMetricsQuery = getInfraMetricsQuery(transaction);
+  const hostName = idx(transaction, _ => _.host.hostname);
+  const podId = idx(transaction, _ => _.kubernetes.pod.uid);
+  const containerId = idx(transaction, _ => _.container.id);
+  const traceId = idx(transaction, _ => _.trace.id);
+  const time = new Date(transaction['@timestamp']).getTime();
```

Comment:
Better to use `transaction.timestamp.us`

Reply:
(note, this will be in microseconds instead of the current millis)

Reply:
Sure! FWIW, just lines moved around.